src/query/functions/src/srfs/mod.rs

```diff
@@ -165,6 +166,68 @@ pub fn register(registry: &mut FunctionRegistry) {
             },
         }))
     });
+
+    registry.properties.insert(
+        "json_array_elements".to_string(),
+        FunctionProperty::default().kind(FunctionKind::SRF),
+    );
+    registry.register_function_factory("json_array_elements", |_, _arg_types| {
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "json_array_elements".to_string(),
+                args_type: vec![DataType::Nullable(Box::new(DataType::Variant))],
```

Comment:
We can check `args_type` first, so that `variant` type doesn't need to be cast to `nullable variant`
for example
```rust
if args_type.len() != 1 || args_type[0].remove_nullable() != DataType::Variant {
    return None;
}
```
you can also reference this example:
https://github.com/datafuselabs/databend/blob/main/src/query/functions/src/srfs/mod.rs#L66-L74